README.md

```diff
@@ -1,60 +1,689 @@
+
+##### d. RabbitBroker (Implements BaseBroker)
+
+**Responsibilities:**
+
+- Implement broker functionalities specific to RabbitMQ.
+
+**Attributes:**
+
+- `connection`: The connection instance with RabbitMQ.
+- `consumers`: Active consumers.
+- `channel`: Communication channel.
+- `session`: Session details.
+- `topic`: Associated topic.
+- `queue`: Associated queue.
+- `rabbitURL`: Connection URL for RabbitMQ.
+
+**Methods:**
+
+- `getConsumers()`: Retrieve active consumers.
+- `getConnection()`: Retrieve the current connection.
+- `consume(eventName, function)`: Subscribe to an event with a callback.
+- `produce(events)`: Emit events to the broker.
+- `closeConnection()`: Terminate the broker connection.
+
+---
+
+### Retry and Dead Letter Queue (DLQ) Components
+
+To support robust retry mechanisms and DLQ functionality internally, the following additional classes and packages are proposed. These components are managed within the framework and are not exposed directly to the external API.
+
+#### 10. RetryManager (Internal Static Class)
+
+**Responsibilities:**
+
+- Manage retry and DLQ policies as well as execution for failed event processing.
+
+**Attributes:**
+
+- `maxAttempts`: Maximum number of retry attempts.
+- `initialDelay`: Initial delay before the first retry (in milliseconds).
+- `maxDelay`: Maximum delay between retries (in milliseconds).
+- `currentAttempt`: Counter for current retry attempt.
+- `dlqName`: Name of the Dead Letter Queue.
+
+**Methods:**
+
+- `shouldRetry()`: Determine if the event should be retried based on `currentAttempt` and `maxAttempts`.
+- `getNextDelay()`: Calculate the next delay to be used in exponential backoff.
+- `incrementAttempt()`: Increase the retry count by one.
+- `reset()`: Reset the retry counter for new events.
+- `moveToDLQ(event)`: Move the failed event to the DLQ.
+
+---
+
+## Behavior and Workflow
+
+### 1. Emitting Events
+
+- **Process:**
+  1. An event is instantiated with a name, payload.
+  2. The `Emitter` triggers the event, sending it to the broker.
+  3. The broker routes the event based on its naming convention to the appropriate queues.
+
+- **Outcome:**
+  - Subscribed listeners receive the event and execute their callback functions.
+
+### 2. Consuming Events
+
+- **Process:**
+  1. A `Listener` subscribes to specific event names or patterns.
+  2. Upon receiving an event, the listener processes it using the defined callback.
+  3. The listener invokes one of the context methods (`success`, `fail`, `reject`) based on the processing outcome.
+
+- **Outcome:**
+  - **Success:** Event is acknowledged and removed from the queue.
+  - **Fail:** Event is requeued for retry, activating retry and DLQ rules.
+  - **Reject:** Event is discarded and routed directly to the DLQ without retrying.
+
+### 3. Running as a Daemon
+
+- **Process:**
+  1. The `Daemon` starts and initializes all event bindings.
```

Comment:
Does each language have its own daemon, or did you create a single interface for all languages to access?